emlib/timer.rs

```diff
@@ -0,0 +1,442 @@
+use core::intrinsics::transmute;
+use core::default::Default;
+
+#[repr(C)]
+pub struct CC {
+    CTRL: u32,
+    CCV: u32,
+    CCVP: u32,
+    CCVN: u32,
+}
+
+#[repr(C)]
+pub struct Timer {
+    CTRL: u32,
+    CMD: u32,
+    STATUS: u32,
+    IEN: u32,
+    IF: u32,
+    IFS: u32,
+    IFC: u32,
+    TOP: u32,
+    TOPB: u32,
+    CNT: u32,
+    ROUTE: u32,
+    RESERVED0: u32,
+    CC: [CC; 3],
+    RESERVED1: [u32; 4],
+    DTCTRL: u32,
+    DTTIME: u32,
+    DTFC: u32,
+    DTOGEN: u32,
+    DTFAULT: u32,
+    DTFAULTC: u32,
+    DTLOCK: u32,
```

Comment:
We should consider defining all struct fields to `pub`. I think it makes more sense to have the fields public, at least for the bindings-library.